cwms-data-api/src/test/java/cwms/cda/data/dao/basin/BasinDaoTest.java

```diff
@@ -0,0 +1,276 @@
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+
+class BasinDaoTest extends DataApiTestIT {
+    private static final String OFFICE_ID = TestAccounts.KeyUser.SWT_NORMAL.getOperatingOffice();
+    private static final String UNIT_SYSTEM = "SI";
+    private static final ArrayList<Location> locationsCreated = new ArrayList<>();
+    private static final Basin parentBasin = new Basin.Builder()
+            .withBasinId(new LocationIdentifier.Builder()
+                    .withName("TEST_PARENT_BASIN")
+                    .withOfficeId(OFFICE_ID)
+                    .build())
+            .withAreaUnit(UNIT_SYSTEM)
+            .withPrimaryStreamId(new LocationIdentifier.Builder()
+                    .withName("TEST_PAR_PRIM_STREAM")
+                    .withOfficeId(OFFICE_ID)
+                    .build())
+            .withTotalDrainageArea(100.0)
+            .withSortOrder(1.0)
+            .build();
+
+    @BeforeAll
+    static void setupTestLocations() {
+        try {
+            createTestLocation("TEST_PAR_PRIM_STREAM", "STREAM");
+            createTestLocation("TEST_PARENT_BASIN", "BASIN");
+            createTestLocation("TEST_BASIN", "BASIN");
+            createTestLocation("TEST_PRIMARY_STREAM", "STREAM");
+
+            CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+            db.connection(c -> {
+                DSLContext ctx = dslContext(c);
+                try {
+                    BasinDao parentBasinDao = new BasinDao(ctx);
+                    parentBasinDao.storeBasin(parentBasin);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterAll
+    public static void cleanUpTestLocations() {
+        Iterator<Location> it = locationsCreated.iterator();
+        while(it.hasNext()) {
+            try {
+                Location location = it.next();
+                CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+                db.connection(c-> {
+                    try(PreparedStatement stmt = c.prepareStatement(deleteLocationQuery)) {
+                        stmt.setString(1,location.getName());
+                        stmt.setString(2,location.getOfficeId());
+                        stmt.execute();
+                    } catch (SQLException ex) {
+                        throw new RuntimeException("Unable to remove location",ex);
+                    }
+                },"cwms_20");
+                it.remove();
+            } catch(SQLException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    private static void createTestLocation(String locationName, String kind) throws SQLException{
```

Comment:
I would use the existing locationDao to do this. I also think you need to specify a Nation and State ("CO") for this.

Reply:
Addressed in latest push